cmd/hub/mcscontroller/controller.go

```diff
@@ -118,15 +121,205 @@ func (r *ServiceExportReconciler) Reconcile(ctx context.Context, req ctrl.Reques
+	es := &mcs.ExportSpec{}
+
+	err := es.UnmarshalObjectMeta(&se.ObjectMeta)
+	if err != nil {
+		return err
+	}
+	namespacedName := types.NamespacedName{
+		Namespace: es.Name,
+		Name:      es.Namespace,
+	}
+	log := r.Log.WithValues("service", namespacedName.Name)
+
+	si := mcsv1a1.ServiceImport{}
+	err = r.Client.Get(ctx, namespacedName, &si)
+	expected := mcsv1a1.ServiceImport{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: es.Namespace,
+			Name:      es.Name,
+		},
+		Spec: mcsv1a1.ServiceImportSpec{
+			Type:  es.Service.Type,
+			Ports: es.Service.Ports,
+		},
+	}
+	if apierrors.IsNotFound(err) {
+		err = r.Client.Create(ctx, &expected)
+		if err != nil {
+			log.Error(err, "unable to create the needed Service Import")
+			return err
+		}
+	} else if err == nil {
+		err = r.updateSi(ctx, &si, &expected, log)
+		if err != nil {
+			return err
+		}
+	} else {
+		log.Error(err, "Failed to get the needed service import")
+		return err
+	}
+	return nil
+}
+
+func (r *ServiceExportReconciler) updateSi(ctx context.Context, si *mcsv1a1.ServiceImport, expected *mcsv1a1.ServiceImport, log logr.Logger) error {
+	//not sure it's really needed to check name and namespace - I got the si from the Get according to those fields
+	//however its a general func, maybe should be "safe" for other usages(?)
+	if si.Name == expected.Name && si.Namespace == expected.Namespace {
+		si.Spec.Type = expected.Spec.Type
+		//maybe add a function that change the ports more carefully, without overwriting
+		si.Spec.Ports = expected.Spec.Ports
+		err := r.Client.Update(ctx, si)
+		if err != nil {
+			log.Error(err, "unable to update the needed Service Import")
+			return err
+		}
+	}
+	return nil
+}
+
+// delete the corresponding serviceImport of the given serviceExport if exists.
+func (r *ServiceExportReconciler) deleteImport(ctx context.Context, se *mcsv1a1.ServiceExport) error {
+	es := &mcs.ExportSpec{}
+
+	err := es.UnmarshalObjectMeta(&se.ObjectMeta)
+	if err != nil {
+		return err
+	}
+	namespacedName := types.NamespacedName{
+		Namespace: es.Name,
+		Name:      es.Namespace,
+	}
+	log := r.Log.WithValues("service", namespacedName.Name)
+
+	si := mcsv1a1.ServiceImport{}
+	err = r.Client.Get(ctx, namespacedName, &si)
+	si = mcsv1a1.ServiceImport{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: es.Namespace,
+			Name:      es.Name,
+		},
+		Spec: mcsv1a1.ServiceImportSpec{
+			Type:  es.Service.Type,
+			Ports: es.Service.Ports,
+		},
+	}
+	if apierrors.IsNotFound(err) {
+		//the serviceImport dosen't exists - don't do anything
+		return nil
+	} else if err == nil {
+		//delete the service Import
+		r.Client.Delete(ctx, &si)
+	} else {
+		log.Error(err, "Failed to get the service Import to delete")
+		return err
+	}
+	return nil
+}
+
+//update the condition field under serviceExport status
+//based on the already existing method in the agent - might move to lhutil
```

Comment:
I think @davidohana may already have something like that in `lhutil`. Please confirm